packages/yup/src/date.js

```diff
@@ -24,28 +28,31 @@ export default class AvDateSchema extends mixed {
   }
 
   typeError() {
-    return this.test({
-      message: 'Date is invalid.',
-      name: 'typeError',
-      test(value) {
-        if (value !== undefined) {
-          if (!this.schema.isType(value)) {
-            // Values that do not pass the previous .isType() check are expected to be a moment object
-            // because this.getValidDate(value) will have run. So as long as the passed in value
-            // is defined, moment._i will contain a string value to validate.
-            // If user enters a date and then removes it, should not show a typeError
-            // Note: this does not prevent other tests, like isRequired, from showing messages
-            // If user has touched a required field, error message should still show
-            return value._i === '';
+    return (
+      this.typeErrorMessage &&
```

Comment:
The `yup` docs say that when `validate` is called (like we do on form submit or onBlur for example) it should always return a valid value or reject with a ValidationError https://github.com/jquense/yup#mixedvalidatevalue-any-options-object-promiseany-validationerror

Since `this.typeErrorMessage` is `undefined` the first time this method is called (it is called when `date.js` calls `super()`), I don't think our test should return `undefined` here. Even though it allows custom messages to work for subsequent validations, it seems wrong that the first run returns `undefined` _and_ requires us to cut a breaking change (users need to chain `.typeError()` to all usages of `avDate()` now) to work.

I think we should try and look for some examples of extending/overriding `typeError` in custom schemas to see if we're on the right track or not.